builders/build/substrate/adding-custom-made-module.md

```diff
@@ -0,0 +1,347 @@
+- **Rust compiler and Cargo package manager**
+
+More about how to install Rust and Cargo is in the [prerequisites article](/builders/build/prerequisites/#installing-rust){target=_blank}.
+
+## Creating the Lottery Module Files {: #creating-lottery-module-files } 
+
+Before you begin writing code, you must create the files that contain the logic. Substrate modules are abstract and designed for reuse in various runtimes with different customizations. To achieve this, you'll use Cargo, which is Rust's package manager, to create the module in the form of a new package.
+
+As mentioned in the prerequisites section, the first step is to clone the [Tanssi repository](https://github.com/moondance-labs/tanssi){target=_blank} and, from the root folder, navigate to `pallets`, where the module will be created.
+
+```bash
+cd container-chains/pallets
+```
+
+Next, create the module package with Cargo:
+
+```bash
+cargo new lottery-example
+```
+
+By default, Cargo creates the new package in a folder with the provided name (`lottery-example`, in this case), containing a manifest file, `Cargo.toml`, and a `src` folder with a `main.rs` file. To respect the naming convention used in Substrate, the `main.rs` file is renamed to `lib.rs`:
+
+```bash
+mv lottery-example/src/main.rs lottery-example/src/lib.rs
+```
+
+Once you've executed all the commands, the module is created and ready to contain the custom logic that you'll be adding in the following sections.
+
+## Configure the Module's Dependencies {: #configure-module-dependencies}
+
+Since the module functions as an independent package, it has its own Cargo.toml file where you must specify the module's attributes and dependencies.
+
+For instance, you can use attributes to specify details like the module's name, version, authors, and other pertinent information. For example, in the the `lottery-example` module, the `Cargo.toml` file can be configured as follows:
+
+```toml
+[package]
+name = "module-lottery-example"
+version = "4.0.0-dev"
+description = "Simple module example"
+authors = [""]
+homepage = ""
+...
+```
+
+This file also defines the module's dependencies, such as the core functionality that allows seamless integration with the runtime and other modules, access to storage, event emission, and more. 
+
+The full example of the `Cargo.toml` file sets, besides the attributes, the dependencies required by Substrate:
+
+??? code "View the complete Cargo.toml file"
+
+    ```rust
+    --8<-- 'code/basic-substrate/lottery-example-cargo.toml'
+    ```
+
+## Adding Custom Logic {: #adding-custom-logic}
+
+As presented in the [custom-made module](/learn/framework/modules/#custom-modules){target=_blank} section of the modularity article, creating a module involves implementing the following attribute macros, of which the first three are mandatory:
+
+- **`#[frame_support::pallet]`** - this attribute is the entry point that marks the module as usable in the runtime
```

Comment:
use snippets if this is already somewhere else

Reply:
I had to rewrite a little the learn article, but now that text is shared content